### PR TITLE
Use QueueNameForObject in Deployment Webhook.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Use QueueNameForObject in Deployment Webhook to validate both QueueName Label and QueueName Annotation. 

#### Which issue(s) this PR fixes:

Prepare for #3555

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/jobs/deployment/deployment_webhook.go [hunk 1]
```diff
@@ -57,16 +57,16 @@ func SetupWebhook(mgr ctrl.Manager, opts ...jobframework.Option) error {
 var _ admission.CustomDefaulter = &Webhook{}
 
 func (wh *Webhook) Default(ctx context.Context, obj runtime.Object) error {
-	d := fromObject(obj)
-	log := ctrl.LoggerFrom(ctx).WithName("deployment-webhook").WithValues("deployment", klog.KObj(d))
+	deployment := fromObject(obj)
+
+	log := ctrl.LoggerFrom(ctx).WithName("deployment-webhook").WithValues("deployment", klog.KObj(deployment))
```
Comment (trasc) [inline pkg/controller/jobs/deployment/deployment_webhook.go:62]:
```suggestion
	log := ctrl.LoggerFrom(ctx).WithName("deployment-webhook")
```
controller-runtime is adding this to the logger before this call.

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve

File: pkg/controller/jobs/deployment/deployment_webhook.go (hunk 1 above)
Comment (mbobrovskyi) [reply pkg/controller/jobs/deployment/deployment_webhook.go:62]: Thanks! I will remove it in separate PR.

**Pull request merged**